src/SqlSquare/Lenses.purs

```diff
@@ -0,0 +1,318 @@
+
+_args ∷ ∀ a r. Lens' { args ∷ a|r } a
+_args = lens _.args _{ args = _ }
+
+_cases ∷ ∀ a r. Lens' { cases ∷ a|r } a
+_cases = lens _.cases _{ cases = _ }
+
+_else ∷ ∀ a r. Lens' { else_ ∷ a|r } a
+_else = lens _.else_ _{ else_ = _ }
+
+_ident ∷ ∀ a r. Lens' { ident ∷ a|r } a
+_ident = lens _.ident _{ ident = _ }
+
+_bindTo ∷ ∀ a r. Lens' { bindTo ∷ a|r } a
+_bindTo = lens _.bindTo _{ bindTo = _ }
+
+_in ∷ ∀ a r. Lens' { in_ ∷ a|r } a
+_in = lens _.in_ _{ in_ = _ } -- __O_M_G__
+
+_isDistinct ∷ ∀ a r. Lens' { isDistinct ∷ a|r } a
+_isDistinct = lens _.isDistinct _{ isDistinct = _ }
+
+_projections ∷ ∀ a r. Lens' { projections ∷ a|r } a
+_projections = lens _.projections _{ projections = _ }
+
+_relations ∷ ∀ a r. Lens' { relations ∷ a|r } a
+_relations = lens _.relations _{ relations = _ }
+
+_filter ∷ ∀ a r. Lens' { filter ∷ a|r } a
+_filter = lens _.filter _{ filter = _ }
+
+_groupBy ∷ ∀ a r. Lens' { groupBy ∷ a|r } a
+_groupBy = lens _.groupBy _{ groupBy = _ }
+
+_orderBy ∷ ∀ a r. Lens' { orderBy ∷ a|r } a
+_orderBy = lens _.orderBy _{ orderBy = _ }
+
+_keys ∷ ∀ a r. Lens' { keys ∷ a|r } a
+_keys = lens _.keys _{ keys = _ }
+
+_having ∷ ∀ a r. Lens' { having ∷ a|r } a
+_having = lens _.having _{ having = _ }
+
+_cond ∷ ∀ a r. Lens' { cond ∷ a|r } a
+_cond = lens _.cond _{ cond = _ }
+
+_alias ∷ ∀ a r. Lens' { alias ∷ a|r } a
+_alias = lens _.alias _{ alias = _ }
+
+_aliasName ∷ ∀ a r. Lens' { aliasName ∷ a|r } a
+_aliasName = lens _.aliasName _{ aliasName = _ }
+
+_left ∷ ∀ a r. Lens' { left ∷ a|r } a
+_left = lens _.left _{ left = _ }
+
+_right ∷ ∀ a r. Lens' { right ∷ a|r } a
+_right = lens _.right _{ right = _ }
+
+_joinType ∷ ∀ a r. Lens' { joinType ∷ a|r } a
+_joinType = lens _.joinType _{ joinType = _ }
+
+_clause ∷ ∀ a r. Lens' { clause ∷ a|r } a
+_clause = lens _.clause _{ clause = _ }
+
+_tablePath ∷ ∀ a r. Lens' { tablePath ∷ a|r } a
+_tablePath = lens _.tablePath _{ tablePath = _ }
+
+
+_SetLiteral
+  ∷ ∀ t
+  . (Recursive t (S.SqlF EJ.EJsonF), Corecursive t (S.SqlF EJ.EJsonF))
+  ⇒ Prism' t (L.List t)
+_SetLiteral = prism' (embed ∘ S.SetLiteral) $ project ⋙ case _ of
+  S.SetLiteral lst → M.Just lst
+  _ → M.Nothing
+
+_Literal
+  ∷ ∀ t
+  . (Recursive t (S.SqlF EJ.EJsonF), Corecursive t (S.SqlF EJ.EJsonF))
+  ⇒ Prism' t (EJ.EJsonF t)
+_Literal = prism' (embed ∘ S.Literal) $ project ⋙ case _ of
+  S.Literal js → M.Just js
+  _ → M.Nothing
+
+_ArrayLiteral
+  ∷ ∀ t
+  . (Recursive t (S.SqlF EJ.EJsonF), Corecursive t (S.SqlF EJ.EJsonF))
+  ⇒ Prism' t (Array t)
+_ArrayLiteral = prism' (embed ∘ S.Literal ∘ EJ.Array) $ project ⋙ case _ of
+  S.Literal (EJ.Array a) → M.Just a
+  _ → M.Nothing
+
+_MapLiteral
+  ∷ ∀ t
+  . (Recursive t (S.SqlF EJ.EJsonF), Corecursive t (S.SqlF EJ.EJsonF))
+  ⇒ Prism' t (Array (t × t))
+_MapLiteral = prism' (embed ∘ S.Literal ∘ EJ.Map) $ project ⋙ case _ of
+  S.Literal (EJ.Map tpls) → M.Just tpls
+  _ → M.Nothing
+
+_Splice
+  ∷ ∀ t
+  . (Recursive t (S.SqlF EJ.EJsonF), Corecursive t (S.SqlF EJ.EJsonF))

```

Comment:
This is a pretty minor thing, but we could make the lenses that aren't literal-specific polymorphic in `SqlF l` too perhaps. I wouldn't hold up this PR for that though!